etrago/cluster/spatial.py

```diff
@@ -656,79 +689,60 @@ def kmedoids_dijkstra_clustering(etrago, buses, connections, weight, n_clusters)
         logger.info(
             "options remove_stubs and use_reduced_coordinates not reasonable for k-medoids Dijkstra Clustering"
         )
-        
-    ### if-condition is only necessary for gas sector, because import of busmap for AC checked before calling of this function
-    ### can be deleted after busmaps are adapted in gas sector aswell
-
-    if ((str(buses.carrier.unique()[0]) == 'CH4' and not settings["k_gas_busmap"]) or (str(buses.carrier.unique()[0]) == 'AC')):
-           
-        ###
-
-        bus_weightings = pd.Series(weight)
-        buses_i = buses.index
-        points = buses.loc[buses_i, ["x", "y"]].values.repeat(
-            bus_weightings.reindex(buses_i).astype(int), axis=0
-        )
 
-        kmeans = KMeans(
-            init="k-means++",
-            n_clusters=n_clusters,
-            n_init=settings["n_init"],
-            max_iter=settings["max_iter"],
-            tol=settings["tol"],
-            random_state=settings["random_state"],
-        )
-        kmeans.fit(points)
+    bus_weightings = pd.Series(weight)
+    buses_i = buses.index
+    points = buses.loc[buses_i, ["x", "y"]].values.repeat(
+        bus_weightings.reindex(buses_i).astype(int), axis=0
+    )
 
-        busmap = pd.Series(
-            data=kmeans.predict(buses.loc[buses_i, ["x", "y"]]),
-            index=buses_i,
-            dtype=object,
-        )
+    kmeans = KMeans(
+        init="k-means++",
+        n_clusters=n_clusters,
+        n_init=settings["n_init"],
+        max_iter=settings["max_iter"],
+        tol=settings["tol"],
+        random_state=settings["random_state"],
+    )
+    kmeans.fit(points)
+
+    busmap = pd.Series(
+        data=kmeans.predict(buses.loc[buses_i, ["x", "y"]]),
+        index=buses_i,
+        dtype=object,
+    )
+
+    # identify medoids per cluster -> k-medoids clustering
+
+    distances = pd.DataFrame(
+        data=kmeans.transform(buses.loc[buses_i, ["x", "y"]].values),
+        index=buses_i,
+        dtype=object,
+    )
+    distances = distances.apply(pd.to_numeric)
+
+    medoid_idx = distances.idxmin()
+
+    # dijkstra's algorithm
+    busmap, busmap_ind = dijkstras_algorithm(
+        buses,
+        connections,
+        medoid_idx,
+        etrago.args["network_clustering"]["CPU_cores"],
+    )
+    busmap.index.name = "bus_id"
+
+    if str(buses.carrier.unique()[0]) == "CH4":
 
-        # identify medoids per cluster -> k-medoids clustering
+        busmap.name = "cluster"
+        busmap_ind.name = "medoid_idx"
 
-        distances = pd.DataFrame(
-            data=kmeans.transform(buses.loc[buses_i, ["x", "y"]].values),
-            index=buses_i,
-            dtype=object,
+        export = pd.concat([busmap, busmap_ind], axis=1)
+        export.index.name = "bus_id"
+        export.to_csv(
+            "kmedoids_gasgrid_busmap_"
```

Comment:
It would be good if you could export the gas busmap in the postprocessing as this function to conduct the k-medoids Dijkstra Clustering is also used in the clustering of the AC-network, hence the busmap with this name would also be created during that process.

Reply:
Done! 
I also removed the return of `busmap_ind` in the dijsktra clustering because it is not used anywhere.